src/Interpreters/ClusterProxy/executeQuery.cpp

```diff
@@ -1068,7 +1069,7 @@ std::optional<QueryPipeline> executeInsertSelectWithParallelReplicas(
 
     for (size_t i = 0; i < connection_pools.size(); ++i)
     {
-        if (skip_local_replica && i == *local_replica_index)
+        if (skip_local_replica && (local_replica_index && i == *local_replica_index))
```

Comment:
I guess we can remove `skip_local_replica`, since it is used just for this purpose - ensure that `local_replica_index` is set